src/MusicalScore/Graphical/MusicSystemBuilder.ts

```diff
@@ -135,7 +135,7 @@ export class MusicSystemBuilder {
             const measureFitsInSystem: boolean = this.currentSystemParams.currentWidth + totalMeasureWidth + nextMeasureBeginInstructionWidth < systemMaxWidth;
             const doXmlPageBreak: boolean = this.rules.NewPageAtXMLNewPageAttribute && sourceMeasure.printNewPageXml;
             const doXmlLineBreak: boolean = doXmlPageBreak || // also create new system if doing page break
-                (this.rules.NewSystemAtXMLNewSystemAttribute && sourceMeasure.printNewSystemXml);
+                (this.rules.NewSystemAtXMLNewSystemAttribute && (sourceMeasure.printNewPageXml || sourceMeasure.printNewSystemXml));
```

Comment:
I would prefer switching the order here, printNewSystem first, printNewPage second. Logically the same of course, but printNewSystem makes more sense so it's more intuitive to come first.

Also, since the printNewPageXml value will only be respected with the new `PageBreakImpliesSystemBreak` EngravingRule, please extract this part of the condition into its own variable, something like:
```ts
const impliedSystemBreak: boolean = sourceMeasure.printNewPageXml && this.rules.PageBreakImpliesSystemBreak;

const doXmlLineBreak: boolean = doXmlPageBreak || // also create new system if doing page break
    (this.rules.NewSystemAtXMLNewSystemAttribute && (sourceMeasure.printNewSystemXml || impliedSystemBreak));
```